### PR TITLE
docs: configs should be directed acyclic graphs

### PR DESCRIPTION
**Pull request by Jasha10**
Followup to https://github.com/omry/omegaconf/pull/602#issuecomment-799542205.

### CODE REVIEW

File: docs/source/usage.rst [hunk 1]
```diff
@@ -382,6 +382,11 @@ Interpolated nodes can be any node in the config, not just leaf nodes:
     >>> (cfg.player.height, cfg.player.weight)
     (180, 75)
 
+Usually the nodes in a config should have the topology of a directed acyclic
+graph. It is possible to use interpolation to create configs that contain
+cycles (i.e. that are not acyclic), but doing so is strongly discouraged and
+may result in undefined behavior.
+
```
Comment (omry) [inline docs/source/usage.rst:389]: We should mention it (more concisely) where we are introducing interpolation (before the examples).

Maybe something like:
```
NOTE: Interpolations may cause config cycles. Such cycles are discouraged and may cause undefined behavior.
```

Comment (Jasha10) [reply docs/source/usage.rst]: Ok, added the new note in https://github.com/omry/omegaconf/pull/604/commits/84bd95b3b23ddea0c95dc6bb2ce38b10fee08eec and removed the old one in https://github.com/omry/omegaconf/pull/604/commits/68713259f9018d6c65081f97578db5ed3c97e97f.

**Review by omry — APPROVED** [review]

File: docs/source/usage.rst (hunk 1 above)
Comment (omry) [reply docs/source/usage.rst]: lgtm, thanks!

**Comment by omry** [comment]
Go ahead and land this :)

**Pull request merged**